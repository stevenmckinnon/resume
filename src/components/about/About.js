import React from 'react';
import personalData from '../../cv-data/PersonalData';

class About extends React.Component {
    render() {
        return (
            <section id="about">
                <div className="row">
                    <div className="three columns">
                        <img className="profile-pic" src="images/steve.jpg" alt="profile"/>
                    </div>
                    <div className="nine columns main-col">
                        <h2>About Me</h2>
                        <p>{personalData.description}</p>
                        <div className="row">
                            <div className="columns contact-details">
                                <h2>Contact Details</h2>
                                <p className="address">
                                    <span>{personalData.name}</span><br />
                                    <span>{personalData.location}</span><br />
                                    <a href={`mailto:${personalData.email}`}> <span>
                                        <i className="fa fa-envelope"></i> Email me!</span></a>
                                </p>
                            </div>
                            <div className="columns download">
                                <p>
                                    <a href={personalData.cvLink} rel="noopener noreferrer" target="_blank" className="button"><i
                                        className="fa fa-download"/>Download CV</a>
                                </p>
                            </div>
                        </div>
                    </div>
                </div>
            </section>
        );
    }
}

export default About;