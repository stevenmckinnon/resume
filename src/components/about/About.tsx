import * as React from 'react';
import GraphImg from "graphcms-image";
import { Element } from 'react-scroll';
import { PersonalDetails, Photo } from '../../interfaces';

interface Props {
    photo: Photo;
    personalDetails: PersonalDetails;
}

const About: React.FC<Props> = ({ photo, personalDetails }) => (
    <Element name="about">
        <section id="about">
            <div className="row about">
                <div className="three columns">
                    <GraphImg className="profile-pic" image={photo} maxWidth={120} />
                </div>
                <div className="nine columns main-col">
                    <h2>About Me</h2>
                    <p>{personalDetails.biography}</p>
                    <div className="row">
                        <div className="columns contact-details">
                            <h2>Contact Details</h2>
                            <p className="contact">
                                <span><b>{personalDetails.name}</b></span><br />
                                <span>{personalDetails.location}</span><br />
                                <a href={`mailto:${personalDetails.email}`}> <span>
                                    <i className="fa fa-envelope"></i> Email me!</span></a>
                            </p>
                        </div>
                        <div className="columns download">
                            <p>
                                <a href={personalDetails.cvUrl} rel="noopener noreferrer" target="_blank"
                                    className="button"><i className="fa fa-download" />Download CV</a>
                            </p>
                        </div>
                    </div>
                </div>
            </div>
        </section>
    </Element>
);

export default About;