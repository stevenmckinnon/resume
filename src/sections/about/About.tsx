import * as React from "react"
import { Element } from "react-scroll"
import { PersonalDetails, Photo } from "../../interfaces"
import { AtSign, Download } from "lucide-react"
import { Button } from "@/components/Button"

interface Props {
  photo: Photo
  personalDetails: PersonalDetails
}

const About: React.FC<Props> = ({ photo, personalDetails }) => (
  <Element name="about" className="bg-primary pt-8 pb-8">
    <section id="about" className="flex justify-center">
      <div className="flex flex-col md:grid grid-cols-3 p-6 max-w-4xl">
        <div className="col-span-1 flex justify-center items-start p-6">
          <img src={photo.url} className="w-40 h-40 rounded-full" alt="profile" />
        </div>
        <div className="col-span-2 p0 md:p-6">
          <h2 className="text-white text-lg font-bold font-heading">About Me</h2>
          <p className="text-white font-body">{personalDetails.biography}</p>
          <div className="m-8">
            <div className="flex flex-col items-center gap-3">
              <h2 className="text-white text-lg font-bold font-heading">
                Contact Details
              </h2>
              <p className="text-white font-body">
                <span>
                  <b>{personalDetails.name}</b>
                </span>
                <br />
                <span>{personalDetails.location}</span>
                <br />
              </p>
              <Button variant="secondary" asChild>
                <a
                  href={`mailto:${personalDetails.email}`}
                  className="flex gap-2 items-center justify-center mt-6 font-body"
                >
                  <AtSign size={16} /> Email me!
                </a>
              </Button>
              <Button variant="secondary" asChild>
                <a
                  href={personalDetails.cvUrl}
                  rel="noopener noreferrer"
                  target="_blank"
                  className="flex gap-2 items-center justify-center font-body"
                >
                  <Download size={16} />
                  Download CV
                </a>
              </Button>
            </div>
          </div>
        </div>
      </div>
    </section>
  </Element>
)

export default About
