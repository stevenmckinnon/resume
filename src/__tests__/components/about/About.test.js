import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import About from '../../../components/about/About';

Enzyme.configure({ adapter: new Adapter() });

describe('Tests the About component', () => {
    it('renders the About component', () => {
        const details = {
            name: 'John Smith',
            location: 'New York, NY',
            biography: 'I am a dev, hear me roar',
            cvUrl: 'http://url.com',
            email: 'hello@email.com'
        };
        const photo = {
            handle: 'handle',
            width: 1333,
            height: 1333
        };
        const wrapper = shallow(<About personalDetails={details} photo={photo} />);
        expect(wrapper).toMatchSnapshot();
    });
});
