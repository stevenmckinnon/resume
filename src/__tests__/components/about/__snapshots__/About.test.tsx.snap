// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the About component renders the About component 1`] = `ShallowWrapper {}`;
